### DISCUSSION TITLE
Question about `reportSum` error and foreign transfers

### DISCUSSION THREAD

**Discussion by antoniomrfranco** [firefly-iii/firefly-iii#9482] [General]
Hello,

I've encountered the following error when starting Firefly III core:

```
  [x]  Error: Transactions for user #1 (REDACTED) are off by -REDACTED!
```

After checking the codebase, I noticed this error is related to a specific `reportSum` check ([here](https://github.com/firefly-iii/firefly-iii/blob/dddaa25d86cc7be4388f46077d05e9be8ab95040/app/Console/Commands/Integrity/ReportSum.php#L69)). From my understanding, this check corresponds to running the following query directly in MariaDB:

```
SELECT SUM(amount) from TRANSACTIONS
```

Upon further investigation, I found that I have many transactions involving transfers to foreign currencies. If I modify the query to include foreign amounts, I get a zero value:

```
SELECT SUM(amount) + SUM(foreign_amount) FROM transactions;
```

It seems like the `reportSum` logic doesn't consider foreign transfers. Shouldn't this be included in the check?

If this is indeed a missing feature or a bug, I’d be happy to help by raising a PR to address it. Let me know if that would be helpful or if there's a preferred approach for resolving this.

Thank you!

**Comment by JC5** [comment]
Hey, thanks for noticing!

This is indeed a bug. Feel free to open a PR!
